apps/snitch_core/lib/core/data/schema/stock/stock_movement.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Snitch.Data.Schema.StockMovement do
+  @moduledoc """
+  Model to track inventory movement in location
+  """
+  use Snitch.Data.Schema
+  use Snitch.Data.Schema.Stock
+
+  @type t :: %__MODULE__{}
+
+  schema "snitch_stock_movements" do
+    field(:quantity, :integer, default: 0)
+    field(:action, :string)
+    field(:originator_type, :string)
+    field(:originator_id, :integer)
+
+    belongs_to(:stock_item, StockItem)
+
+    timestamps()
+  end
+
+  @quantity_limits %{min: round(:math.pow(-2, 31)), max: round(:math.pow(2, 31) - 1)}
+  @create_fields ~w(quantity stock_item_id)a
+
+  def create_fields, do: @create_fields
+
+  @doc """
+  Stock Movements and Stock Transfers are only created.
+  Inorder to update them we can create another stock movement to reverse its effects.
+  """
+  @spec changeset(__MODULE__.t(), map, atom) :: Ecto.Changeset.t()
+  def changeset(instance, params, operation \\ :create)
+  def changeset(instance, params, :create), do: do_changeset(instance, params, @create_fields)
```

Comment:
The `:create` is required because of the QueryHelper, I believe? Then there's no need to make the `operation` optional -- who else will call this changeset function?